File: tests/journal.lib/debug_shared.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// type aliases
using trash_t = pyre::journal::trash_t;
using channel_t = pyre::journal::debug_t;


// verify that channels that have the same name share the same state
int main() {
    // make a channel
    channel_t ch_1("tests.journal");
    // flip all state away from the defaults
    ch_1.active(true);
    ch_1.fatal(true);
    // install a trash can as the channel device
    ch_1.device<trash_t>();

    // make another with the same name
    channel_t ch_2("tests.journal");
    // verify that it sees the common state
    assert(ch_2.active() == ch_1.active());
    assert(ch_2.fatal() == ch_1.fatal());
    assert(ch_2.device() == ch_1.device());

    // all done
    return 0;
}


// end of file

File: tests/journal.ext/debug_shared.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that channels with the same name have common state
    """
    # access
    from journal import libjournal

    # make a channel
    ch_1 = libjournal.Debug("test.channel")
    # activate it
    ch_1.active = True
    # and make it fatal
    ch_1.fatal = True

    # make another
    ch_2 = libjournal.Debug("test.channel")
    # verify it is active
    assert ch_2.active is True
    # and fatal
    assert ch_2.fatal is True
    # deactivate it
    ch_2.active = False
    # and make it non-fatal
    ch_2.fatal = False

    # verify that both channels are now inactive
    assert ch_1.active is False
    assert ch_2.active is False
    # and once again, using {__bool__}
    assert bool(ch_1) is False
    assert bool(ch_2) is False

    # verify that they are both non-fatal
    assert ch_1.fatal is False
    assert ch_2.fatal is False

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
